plugins/huaweicloud/src/main/java/org/eclipse/xpanse/plugins/huaweicloud/manager/HuaweiCloudManagerService.java

```diff
@@ -0,0 +1,104 @@
+ */
+@Slf4j
+@Component
+public class HuaweiCloudManagerService {
+
+    private final CredentialCenter credentialCenter;
+    private final HuaweiCloudManagerClient huaweiCloudManagerClient;
+    private final HuaweiCloudManagerConverter huaweiCloudManagerConverter;
+
+    /**
+     * Constructs a HuaweiCloudManagerService with the necessary dependencies.
+     */
+    public HuaweiCloudManagerService(
+            CredentialCenter credentialCenter,
+            HuaweiCloudManagerClient huaweiCloudManagerClient,
+            HuaweiCloudManagerConverter huaweiCloudManagerConverter) {
+        this.credentialCenter = credentialCenter;
+        this.huaweiCloudManagerClient = huaweiCloudManagerClient;
+        this.huaweiCloudManagerConverter = huaweiCloudManagerConverter;
+    }
+
+    /**
+     * Start the Huawei Cloud Ecs server.
+     */
+    public boolean startService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchStartServersRequest request =
+                huaweiCloudManagerConverter.buildBatchStartServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchStartServersResponse batchStartServersResponse = ecsClient.batchStartServers(request);
+        if (HttpStatus.OK.value() == batchStartServersResponse.getHttpStatusCode()) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Stop the Huawei Cloud Ecs server.
+     */
+    public boolean stopService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchStopServersRequest batchStopServersRequest =
+                huaweiCloudManagerConverter.buildBatchStopServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchStopServersResponse batchStopServersResponse =
+                ecsClient.batchStopServers(batchStopServersRequest);
+        if (HttpStatus.OK.value() == batchStopServersResponse.getHttpStatusCode()) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Restart the Huawei Cloud Ecs server.
+     */
+    public boolean restartService(ServiceManagerRequest serviceManagerRequest) {
+        EcsClient ecsClient = getEcsClient(serviceManagerRequest);
+        BatchRebootServersRequest request =
+                huaweiCloudManagerConverter.buildBatchRebootServersRequest(
+                        serviceManagerRequest.getDeployResourceEntityList());
+        BatchRebootServersResponse response = ecsClient.batchRebootServers(request);
+        if (HttpStatus.OK.value() == response.getHttpStatusCode()) {
+            return true;
+        }
+        return false;
+    }
+
+    private EcsClient getEcsClient(ServiceManagerRequest serviceManagerRequest) {
+        AbstractCredentialInfo credential =
```

Comment:
here must also consider the type of the service. If the service is "vendor hosted" then we must use the credentials from the vendor.